generators/repo/src/commands/generate.ts

```diff
@@ -284,42 +285,43 @@ export class GenerateCommand implements RepomanCommand {
 
     private writeResultsFile = async (results: RemotePushResult[]) => {
         return new Promise<void>(async (resolve, reject) => {
-            const pushedResults = results.filter(r => r.hasChangesFromBase);
+            var comittedResults: RemotePushResult[] = results.filter(r => r.hasChangesFromBase);
+            var mergedResults: RemotePushResult[] = results.filter(r => r.hasChanges);
 
-            if (pushedResults.length === 0 || !this.options.resultsFile) {
+            if (comittedResults.length === 0 && mergedResults.length === 0 || !this.options.resultsFile) {                
                 return resolve();
             }
 
+            const pushedResults = comittedResults.length === 0 ? mergedResults:comittedResults;
             const resultsFilePath = path.resolve(path.normalize(this.options.resultsFile));
-            const resultsFile = await fs.open(resultsFilePath, "a+");
+
+            let fileContent=[];
+            if(existsSync(resultsFilePath)){
+                const fileReadBuffer = await fs.readFile(resultsFilePath);
+                const fileContentJson = fileReadBuffer.toString();
+                if(fileContentJson){
+                    fileContent = JSON.parse(fileContentJson); 
+                }
+            }
+
+            const resultsFile = await fs.open(resultsFilePath, "w+");
             const resultsStream = resultsFile.createWriteStream();
 
             try {
-
-                const output: string[] = [];
-                output.push(`### Project: **${this.manifest.metadata.name}**`);
-
-                for (const result of pushedResults) {
-                    output.push(`#### Remote: **${result.remote}**`);
-                    output.push(`##### Branch: **${result.branch}**`);
-                    output.push('');
-                    output.push('You can initialize this project with:');
-                    output.push('```bash');
-                    output.push(`azd init -t ${result.org}/${result.repo} -b ${result.branch}`);
-                    output.push('```');
-                    output.push('');
-                    output.push(`[View Changes](${result.branchUrl}) | [Compare Changes](${result.compareUrl})`);
-                    output.push('');
-                    output.push('---');
-                    output.push('');
+                const generatedResults = {
+                    metadataName:  this.manifest.metadata.name,
+                    results: pushedResults
```

Comment:
This is a great change, thanks! Will allow for additional integration opportunities exposing the raw repoman json output vs only ever generating a markdown file.